recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.tmpl

```diff
@@ -761,17 +779,18 @@ def main(argv):
         ====================================
 Hit the Enter Key to Exit:
         '''
-        if not "No firmware image could be updated on this board" == str(e):
+        if str(e) != ("OS image version must be newer than the minimal "
+            "version, no firmware image could be updated on this board!"):
```

Comment:
This is too fragile, better to use something more solid, such as inherited error type.